plugins/qeta-backend/src/service/upload/azureBlobStorage.ts

```diff
@@ -0,0 +1,73 @@
+import * as fs from 'fs';
+import { v4 as uuidv4 } from 'uuid';
+import { Config } from '@backstage/config';
+import { QetaStore } from '../../database/QetaStore';
+import { Attachment } from '@drodil/backstage-plugin-qeta-common';
+import { File } from '../types';
+import { getAzureBlobServiceClient } from '../util';
+import { AttachmentStorageEngine, AttachmentStorageEngineOptions } from './attachmentStorageEngine';
+
+class AzureBlobStorageEngine implements AttachmentStorageEngine {
+  config: Config;
+  database: QetaStore;
+  backendBaseUrl: string;
+  qetaUrl: string;
+  container: string;
+
+  constructor(opts: AttachmentStorageEngineOptions) {
+    this.config = opts.config;
+    this.database = opts.database;
+    this.backendBaseUrl = this.config.getString('backend.baseUrl');
+    this.qetaUrl = `${this.backendBaseUrl}/api/qeta/attachments`;
+    this.container =
+      this.config.getOptionalString('qeta.storage.blobStorageContainer') ||
+      'backstage-qeta-images';
+  }
+
+  handleFile = async (
+    file: File,
+    options?: { postId?: number; answerId?: number; collectionId?: number },
+  ): Promise<Attachment> => {
+    const imageUuid = uuidv4();
+    const filename = `image-${imageUuid}-${Date.now()}.${file.ext}`;
+
+    const imageURI = `${this.qetaUrl}/${imageUuid}`;
+    const client = getAzureBlobServiceClient(this.config);
+    const container = client.getContainerClient(this.container);
+    if (!(await container.exists())) {
+        await container.create();
+    }
+
+    await container.uploadBlockBlob(filename, fs.createReadStream(file.path), file.size);
+
+    return await this.database.postAttachment({
+      uuid: imageUuid,
+      locationType: 'azure',
+      locationUri: imageURI,
+      extension: file.ext,
+      mimeType: file.mimeType,
+      path: filename,
+      binaryImage: Buffer.from('0'),
+      creator: '', // required to run locally on sqlite, otherwise it complains about null.
+      ...options,
+    });
+  };
+
+  getAttachmentBuffer = async (attachment: Attachment) => {
+    const client = getAzureBlobServiceClient(this.config);
+    const container = client.getContainerClient(this.container);
+    const blob = container.getBlockBlobClient(attachment.path);
```

Comment:
What happens if any of these do not exist? Does the client throw an error?

Reply:
If I delete the blob from the storage account manually I get back a 404 when I do a get on the attachment. However, it does expose some details that we might not want to expose to the client, so it probably makes sense to add a try/catch here.

Reply:
I can add a `if (await blob.exists())` here to make sure that the blob exists and return `undefined` if it does not. And do the same for the `container` as well I guess.

That does result in a 500 now though, since the existing logic in `attachments.ts` returns that status code in case the `imageBuffer` is undefined. Do we want to change that to a 404?

Reply:
404 would be the right response but not sure all engines work the same way with undefined return value 🤔

Reply:
I've made sure that the other implements adhere to the same semantics and changed `attachments.ts` to return a 404 if no actual image could be found.